tdms/src/matrix_collection.cpp

```diff
@@ -39,5 +42,8 @@ MatFileMatrixCollection::~MatFileMatrixCollection() {
   if (mat_file != nullptr){
     matClose(mat_file);
   }
-  mxFree(matrix_names);
+  std::vector<char*> matrix_names_cstrs{};
```

Comment:
Can I ask what's going on here? Why can't we just
```c++
matrix_names.clear()
```
to deallocate the memory?

Reply:
Ah, no it's just me converting the `std::vector<std::string>` back to an array of char* (via an `std::vector<char*>`). Because that's what `mxFree` wants.

Reply:
Oh wait, that's what [`mxFree` does](https://uk.mathworks.com/help/matlab/apiref/mxfree.html). I get your question. 


The answer isL code-blindness. I translated without thinking.

Can probably delete all of this and leave the vector to go out of scope on class destruction.